liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -1563,6 +1571,89 @@ public void execute() throws MojoExecutionException {
+    protected List<MavenProject> getMultiModuleDevModeProjects(ProjectDependencyGraph graph) throws MojoExecutionException {
+    	getLog().debug("Resolve dev mode projects");
+    	
+        List<MavenProject> sortedReactorProjects = graph.getSortedProjects();
+        Set<MavenProject> conflicts = new LinkedHashSet<MavenProject>(); // keeps the order of items added in
+
+        // A leaf here is a module without any downstream modules depending on it
+        List<MavenProject> leaves = new ArrayList<MavenProject>();
+        for (MavenProject reactorProject : sortedReactorProjects) {
+            if (graph.getDownstreamProjects(reactorProject, true).isEmpty()) {
+                leaves.add(reactorProject);
+                getLog().debug("Found final downstream project: " + reactorProject.getArtifactId());
+            }
+        }
+        
+        // Remove jar projects since these are not relavent to dev mode
+        List<MavenProject> jarLeaves = new ArrayList<MavenProject>();
+        for (MavenProject leaf : leaves) {
+        	if (leaf.getPackaging().equals("jar")) {
+        		jarLeaves.add(leaf);
+        		getLog().debug("Removing jar project: " + leaf.getArtifactId());
+        	}
+        }
+        leaves.removeAll(jarLeaves);
+        
+        // Remove any projects that are configured to be skipped
+        List<MavenProject> skipLeaves = new ArrayList<MavenProject>();
+        for (MavenProject leaf : leaves) {
+        	
+            // Properties that are set in the pom file
+            Properties props = leaf.getProperties();
+
+            // Properties that are set by user via CLI parameters
+            Properties userProps = session.getUserProperties();
+        	Plugin libertyPlugin = getLibertyPluginForProject(leaf);
+            Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(libertyPlugin, "dev", getLog());
+            
+            boolean skipGoal = DevHelper.getBooleanFlag(config, userProps, props, "skip");
+            
+        	if (skipGoal) {
+        		getLog().debug("Skip configured on project: " + leaf.getArtifactId());
+        		skipLeaves.add(leaf);
+        	}
+        }
+        leaves.removeAll(skipLeaves);
+        
+        // Find any remaining conflicts
+        for (MavenProject leaf1 : leaves) {
+            for (MavenProject leaf2 : leaves) {
+                if (leaf1 != leaf2 && !(isSubModule(leaf2, leaf1) || isSubModule(leaf1, leaf2)) ) {
+                    conflicts.add(leaf1);
+                    conflicts.add(leaf2);
+                }
+            }
+        }
+        
+        if (conflicts.isEmpty() ) {
+        	List<MavenProject> devModeProjects = new ArrayList<MavenProject>();
+        	for (MavenProject leaf : leaves) {
+        		devModeProjects.addAll(graph.getUpstreamProjects(leaf, true));
+        		devModeProjects.add(leaf);
+        	}
+        	
+        	for (MavenProject project : devModeProjects) {
+        		getLog().debug("Dev mode projects list: ");
```

Comment:
Do you want this line repeated before every project artifactId? Or just once before a list of all the artifact ids?